src/main/java/com/cloudbees/jenkins/plugins/bitbucket/hooks/WebhookAutoRegisterListener.java

```diff
@@ -186,6 +225,30 @@ private void removeHooks(SCMSourceOwner owner) throws IOException, InterruptedEx
         }
     }
 
+    private BitbucketApi bitbucketApiFor(BitbucketSCMSource source) {
+        switch (new BitbucketSCMSourceContext(null, SCMHeadObserver.none())
+                .withTraits(source.getTraits())
```

Comment:
traits are not supposed to do network things when being applied to the context... they decorate the request for those things